src/combinators.js

```diff
@@ -428,7 +430,11 @@ class ProxyArrow extends Arrow {
     }
 
     isAsync() {
-        return this.ensureFrozen(a => a.isAsync());
+        if (this._isAsync === undefined) {
```

Comment:
Is this just to cache the result? Why the false assignment on 434?

Reply:
So this would work:
Arrow.fix(a => (x => x).lift().seq(a))

It assume it's not async first, then set it correctly if some other parts are async.